### DISCUSSION TITLE
como rastrear seu pacote no NPM?

### DISCUSSION THREAD

**Discussion by vitebo** [frontendbr/forum#2040] [Perguntas e Respostas] [answered]
eai pessoal, tudo bem?

Dando um pouco de contexto...
hoje existe um pacote privado no NPM sendo usado por diversos projetos na empresa onde eu trabalho. Existe alguma forma de rastrear quais são os projetos que utilizam esse pacote?

procurando no NPM não consegui encontrar nada mas me parece ser algo bem comum, principalmente para pacotes privados. Alguém ja passou por isso alguma vez? ou tem ideia de como contornar a situação? qualquer ajuda é super bem vinda 😄 

---

A ideia que eu tenho em mente é utilizando a API do github, buscar todos os projetos da empresa que possuem um `package.json` e depois fazer um find pelo nome do pacote.

Parece funcionar mas é um pouquinho trabalhoso, por isso seria legal saber se alguém conhece alguma lib do gênero ou algo nativo no NPM que consiga ajudar. 

PS: Esse pacote privado também pertence a empresa.

**Comment by ninetails** [comment, marked as answer]
Se for no site do npm, ele próprio tem uma aba de "Dependents", mas creio que ele só checa quando os pacotes estão também no mesmo registro (que é o deles).

Mas se for em registros privados, creio que só fazendo o grafo manualmente... Dá pra baixar todos os projetos da tua empresa e dar um `npm/yarn list <nome do pacote>`

**Reply by vitebo** [reply]
isso, a aba "Dependents" funciona mas só para os casos onde o projeto que esta utilizando esse pacote tbm esteja publicado no npm _(o que não é verdade na maior parte dos casos)_.

Mas seguindo essa sua ideia, no meu caso como todos os repos estão dentro de uma mesma org do github, vou ver se esses `packages` do github conseguem de alguma forma linkar esses caras

**Reply by ninetails** [reply]
Então, depois do `install` dá pra saber quais pacotes estão sendo usados via [npm ls <nome do pacote>](https://docs.npmjs.com/cli/v7/commands/npm-ls) ou [yarn list <nome do pacote>](https://classic.yarnpkg.com/en/docs/cli/list/), mas isso que falei, muito provavelmente o inverso pode não ser possível.

Também desconheço uma ferramenta que faça tal grafo de forma inversa :\

Se for um _monorepo_ até pode ter alguma ferramenta, mas caso não for não sei de uma forma mais automatizada que _text search_.
